lib/SILOptimizer/Mandatory/TFConstExpr.cpp

```diff
@@ -202,9 +202,12 @@ SymbolicValue ConstExprFunctionState::computeConstantValue(SILValue value) {
 
   // If this a trivial constant instruction that we can handle, then fold it
   // immediately.
-  if (isa<IntegerLiteralInst>(value) || isa<FloatLiteralInst>(value) ||
-      isa<StringLiteralInst>(value))
-    return SymbolicValue::getConstantInst(cast<SingleValueInstruction>(value));
+  if (auto *ili = dyn_cast<IntegerLiteralInst>(value))
+    return SymbolicValue::getInteger(ili->getValue(), evaluator.getAllocator());
```

Comment:
@lattner Adding a place holder here to continue our discussion on 64bit vs 2048bit APints. Here I believe we are preserving the bitwidth of the APInt in the inst (though it's making a copy via allocator, which seems needed in the general case since the inst could go away). Does this look reasonable?